src/basic/process-util.c

```diff
@@ -1597,6 +1597,23 @@ int pidfd_verify_pid(int pidfd, pid_t pid) {
         return current_pid != pid ? -ESRCH : 0;
 }
 
+int pidfd_is_alive(int pidfd) {
+        pid_t pid;
+        int r, ret;
+
+        r = pidfd_get_pid(pidfd, &pid);
+        if (r < 0)
+                return r;
+
+        ret = pid_is_alive(pid);
+
+        r = pidfd_verify_pid(pidfd, pid);
```

Comment:
This is pointless. In `pidfd_verify_pid` we ask the pidfd for the pid to compare, which is where the pid originates from...

Reply:
@poettering explicitly requested this behavior I think ...

> please add pidfd_is_alive() that first translates pidfd to pid internally, and then does the pid_is_alive() check via the pid, and then checks if the pid still matches the pidfd afterwards, before returning the result. So that the pid cannot be reused while we operate on it

Reply:
Oops, sorry. Ignore me then.

Reply:
Hmm, maybe some explanatory comments could be added?

Reply:
I've added the comment